buildSrc/src/test/java/com/osacky/flank/gradle/integration/SanityRoboCheck.kt

```diff
@@ -0,0 +1,328 @@
+      |  disable-sharding: false
+      |  smart-flank-disable-upload: false
+      |  local-result-dir: overrideDir
+      |  legacy-junit-result: false
+      |  full-junit-result: false
+      |  output-style: single
+    """
+    }
+  }
+
+  @Test
+  fun checkSanityRoboRunRoboScript() {
+    makeGradleFile(where = testProjectRoot) {
+      """
+      |plugins {
+      |  id "com.osacky.fladle"
+      |}
+      |
+      |fladle {
+      |  serviceAccountCredentials = layout.projectDirectory.file("flank-gradle-service.json")
+      |  debugApk = "foo.apk"
+      |  roboScript = "some/path/script.json"
+      |}
+    """
+    }
+
+    val result = gradleRun {
+      arguments = listOf("writeConfigProps", "-PsanityRobo")
+      projectDir = testProjectRoot.root
+    }
+
+    assertThat(result.output).contains("SUCCESS")
+
+    "build/fladle/flank.yml" readAndCompareWith {
+      """
+      |gcloud:
+      |  app: foo.apk
+      |  device:
+      |  - model: NexusLowRes
+      |    version: 28
+      |
+      |  use-orchestrator: false
+      |  auto-google-login: false
+      |  record-video: true
+      |  performance-metrics: true
+      |  timeout: 15m
+      |  num-flaky-test-attempts: 0
+      |
+      |flank:
+      |  keep-file-path: false
+      |  ignore-failed-tests: false
+      |  disable-sharding: false
+      |  smart-flank-disable-upload: false
+      |  legacy-junit-result: false
+      |  full-junit-result: false
+      |  output-style: single
+    """
+    }
+  }
+
+  @Test
+  fun checkSanityRoboRunRoboDirectives() {
+    makeGradleFile(where = testProjectRoot) {
+      """
+      |plugins {
+      |  id "com.osacky.fladle"
+      |}
+      |
+      |fladle {
+      |  serviceAccountCredentials = layout.projectDirectory.file("flank-gradle-service.json")
+      |  debugApk = "foo.apk"
+      |  roboDirectives = [
+      |    ["click", "button1", ""],
+      |    ["ignore", "button2"],
+      |    ["text", "field1", "my text"],
+      |  ]
+      |}
+    """
+    }
+
+    val result = gradleRun {
+      arguments = listOf("writeConfigProps", "-PsanityRobo")
+      projectDir = testProjectRoot.root
+    }
+
+    assertThat(result.output).contains("SUCCESS")
+
+    "build/fladle/flank.yml" readAndCompareWith {
+      """
+      |gcloud:
+      |  app: foo.apk
+      |  device:
+      |  - model: NexusLowRes
+      |    version: 28
+      |
+      |  use-orchestrator: false
+      |  auto-google-login: false
+      |  record-video: true
+      |  performance-metrics: true
+      |  timeout: 15m
+      |  num-flaky-test-attempts: 0
+      |
+      |flank:
+      |  keep-file-path: false
+      |  ignore-failed-tests: false
+      |  disable-sharding: false
+      |  smart-flank-disable-upload: false
+      |  legacy-junit-result: false
+      |  full-junit-result: false
+      |  output-style: single
+    """
+    }
+  }
+
+  private infix fun String.readAndCompareWith(block: () -> String) = testProjectRoot
```

Comment:
can we rename this to `contains`? it is confusing if the method is named `compareWith` but the assertion is performing `contains`

Also to simplify the comparison, you can use the `printYml` task. It will already read the `yml` file that was written to disk so you can just check the gradle build output.